internal/config/config.go

```diff
@@ -209,3 +221,44 @@ func stringToSliceHookFunc() mapstructure.DecodeHookFunc {
 		return strings.Fields(raw), nil
 	}
 }
+
+// trie is a prefix trie used to create a map of potential
+// key paths from the current environment.
+// This is used to support pre-emptively binding env vars
+// for the keys of map[string]T which we don't know aheaad of
+// time.
+type trie map[string]trie
+
+func (t trie) getPrefix(parts ...string) (trie, bool) {
```

Comment:
could we add some more tests around this method and the function below it if possible?

Reply:
I've taken another pass in a slightly different way. Which also allowed me to test it more thoroughly.
It also accounts for something this implementation couldn't do. Which was keys in maps with `_` in them.
See what you think.